lib/operations/mongo_client_ops.js

```diff
@@ -12,9 +12,94 @@ const ReplSet = require('../topologies/replset');
+  maxIdleTimeMS: { type: 'number' },
+  maxPoolSize: { type: 'number', default: 5, alias: 'poolSize' },
+  poolSize: { type: 'number' },
+  maxStalenessSeconds: { type: 'number' },
+  readConcernLevel: { type: 'string', alias: 'readConcern.level' },
+  readPreference: { type: [ReadPreference, 'object', 'string'] },
+  readPreferenceTags: { type: 'object' },
+  replicaSet: { type: 'string' },
+  retryWrites: { type: 'boolean', default: false },
+  serverSelectionTimeoutMS: { type: 'number', default: 30000 },
+  serverSelectionTryOnce: { type: 'boolean', default: true },
+  socketTimeoutMS: { type: 'number', default: 360000 },
+  ssl: { type: 'boolean', default: false },
+  tls: { type: 'boolean', default: false },
+  tlsAllowInvalidCertificates: { type: 'boolean', default: false },
+  tlsAllowInvalidHostnames: { type: 'boolean', default: false },
+  tlsCAFile: { type: 'string' },
+  tlsCertificateKeyFile: { type: 'string' },
+  tlsCertificateKeyFilePassword: { type: 'string' },
+  tlsInsecure: { type: 'boolean', default: false },
+  w: { type: 'number' },
+  wtimeout: { type: 'number', alias: 'wTimeoutMS' },
+  zlibCompressionLevel: { type: 'number', default: -1 },
+  // TODO: check these old options
+  sslCA: { type: 'string' },
+  sslCert: { type: 'string' },
+  sslValidate: { type: 'boolean', default: true },
+  sslKey: { type: 'buffer' },
+  sslPass: { type: 'string' },
+  sslCRL: { type: 'buffer' },
+  autoReconnect: { type: 'boolean', default: true },
+  auto_reconnect: { type: 'boolean', default: true },
+  noDelay: { type: 'boolean', default: true },
+  keepAlive: { type: 'boolean', default: true },
+  keepAliveInitialDelay: { type: 'number', default: 30000 },
+  family: { type: 'number' },
+  reconnectTries: { type: 'number', default: 30 },
+  reconnectInterval: { type: 'number', default: 1000 },
+  ha: { type: 'boolean', default: true },
+  haInterval: { type: 'number', default: 10000 },
+  acceptableLatencyMS: { type: 'number', default: 15 },
+  connectWithNoPrimary: { type: 'boolean', default: false },
+  forceServerObjectId: { type: 'boolean', default: false },
+  serializeFunctions: { type: 'boolean', default: false },
+  ignoreUndefined: { type: 'boolean', default: false },
+  raw: { type: 'boolean', default: false },
+  bufferMaxEntries: { type: 'number', default: -1 },
+  pkFactory: { type: ['object', 'function'] },
+  promiseLibrary: { type: ['object', 'function'] },
+  readConcern: { type: 'object' },
+  'readConcern.level': { type: 'string' },
+  loggerLevel: { type: 'string' },
+  logger: { type: ['object', 'function'] },
+  promoteValues: { type: 'boolean' },
+  promoteBuffers: { type: 'boolean' },
+  promoteLongs: { type: 'boolean' },
+  domainsEnabled: { type: 'boolean', default: false },
+  validateOptions: { type: 'boolean', default: false, deprecated: 'unknownOptionsWarningLevel' },
+  checkServerIdentity: { type: ['boolean', 'function'], default: true },
+  'auth.user': { type: 'string' },
```

Comment:
I didn't realize we supported dot notation in our schemas. Where do we handle this?

Reply:
Actually we don't support this, good catch.

Reply:
looks like you still need to remove it here